metadata-foundationdb/CMakeLists.txt

```diff
@@ -14,17 +14,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_library(hyperwarp-metadata-fdb SHARED metadata-foundationdb.c)
-add_custom_command(TARGET hyperwarp-metadata-fdb PRE_BUILD COMMAND astyle ARGS "${CMAKE_CURRENT_SOURCE_DIR}/*.c,*.h")
-target_link_libraries(hyperwarp-metadata-fdb fdb_c)
-target_link_libraries(hyperwarp-metadata-fdb m)
-target_link_libraries(hyperwarp-metadata-fdb rt)
-target_link_libraries(hyperwarp-metadata-fdb hyperwarp-metadata-model)
-target_include_directories(hyperwarp-metadata-fdb PUBLIC
+add_library(metadata-foundationdb MODULE metadata-foundationdb.c)
+add_custom_command(TARGET metadata-foundationdb PRE_BUILD COMMAND astyle ARGS "${CMAKE_CURRENT_SOURCE_DIR}/*.c,*.h")
+target_link_libraries(metadata-foundationdb fdb_c)
+target_link_libraries(metadata-foundationdb m)
+target_link_libraries(metadata-foundationdb rt)
+target_link_libraries(metadata-foundationdb hyperwarp-metadata)
+target_link_libraries(metadata-foundationdb hyperwarp-metadata-model)
+target_include_directories(metadata-foundationdb PUBLIC
                           "${PROJECT_BINARY_DIR}"
                           "${PROJECT_SOURCE_DIR}"
                           "${CMAKE_CURRENT_SOURCE_DIR}"
                           )
-
-set_target_properties(hyperwarp-metadata-fdb PROPERTIES PUBLIC_HEADER metadata-foundationdb.h)
-install(TARGETS hyperwarp-metadata-fdb LIBRARY PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hyperwarp)
+set_target_properties(metadata-foundationdb PROPERTIES PREFIX "")
+install(TARGETS metadata-foundationdb LIBRARY DESTINATION ${PROJECT_BINARY_DIR})
```

Comment:
I think we need to revisit whether `${PROJECT_BINARY_DIR}` is the right value here for ´make install´.

Reply:
I'm open to suggestions. right now I never do make install.

Reply:
what about `${CMAKE_INSTALL_LIBDIR}/hyperwarp` which would result in something like `/usr/local/lib/hyperwarp`?

Reply:
sounds good